### PR TITLE
release v0.6.1

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This adds the CHANGELOG for the `v0.6.1` patch release. Also adds a "patch release" section to RELEASE.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +22,62 @@
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+# v0.6.1
+
+This is a patch release that predominantly includes updated performance tests
```
Comment (sunjayBhatia) [inline CHANGELOG.md:27]:
```suggestion
This is a patch release that predominantly includes updated conformance tests
```

Comment (shaneutt) [reply CHANGELOG.md]: Good catch thank you :) a83679beeb56d7789dedd373fb23605e46b3b6c9

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +22,66 @@
...
+  provide a more realistic test for implementations.
+  (#1655, @arkodg)
+- A "SkipTests" field has been added to accomodate implementations in
+  running subsets of the tests as needed, this can be particularly helpful
+  for new implementations that want to add conformance iteratively.
+  (#1578, @mlavacca)
+- ParentRefNotPermitted added to support situations where ReferenceGrant
+  is not allowed due to lack of a ReferenceGrant in the destination NS.
+  (#1672, @Sajiyah-Salat)
+- Fixed a broken test for GRPCRoute that caused an erronous failure.
```
Comment (skriss) [inline CHANGELOG.md:71]: Noting https://github.com/kubernetes-sigs/gateway-api/pull/1692#discussion_r1099010300 here, would be good to get fixed before the release.

Comment (shaneutt) [reply CHANGELOG.md:71]: Yep, thank you @skriss we'll hold on making sure that's sorted out.

Comment (shaneutt) [reply CHANGELOG.md:68]: I've created https://github.com/kubernetes-sigs/gateway-api/issues/1703 and it will be considered a blocker for this PR.

**Comment by shaneutt** [comment]
@robscott, @youngnick, @skriss: tests are passing and we've resolved the things reported yesterday, so we should be good for review.

File: RELEASE.md [hunk 3]
```diff
@@ -37,15 +37,32 @@ release.
...
+- Update `pkg/generator/main.go` with the new semver tag and any updates to the API review URL.
+- Run the following command `BASE_REF=vmajor.minor.patch make generate` which will update generated docs
+  and webhook with the correct version info. Note that the YAMLs will not work until the tag is actually
+  published in the next step.
+- Create a pull request of the `<githubuser>/release-x.x.x` branch into the `release-x.x` branch upstream
+  (which should already exist since this is a patch release).
+- Verify the CI tests pass and once the above merges publish a new Git tag. This can be done using the
+  `git` CLI or Github's [release][release] page.
+- Run the `make build-install-yaml` command which will generate
+  install files in the `release/` directory
```
Comment (mlavacca) [inline RELEASE.md:53]:
```suggestion
  install files in the `release/` directory.
```

Comment (shaneutt) [reply RELEASE.md]: :+1: 37b52e2daf897f6b9b0b8d360512c26e60dbac8e updates

**Review by mlavacca — APPROVED** [review]

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]

**Comment by robscott** [comment]
Thanks @shaneutt! LGTM as well

**Pull request merged**